py3status/modules/diskdata.py

```diff
@@ -79,6 +87,39 @@ class Py3status:
     }
     unit = "B/s"
 
+    class Meta:
+        deprecated = {
+            "remove": [{"param": "sector_size", "msg": "obsolete parameter"}],
+            "rename": [
+                {"param": "disk", "new": "disks", "msg": "rename parameter to `disks`"}
+            ],
+        }
+
+    def _resolve_disk_name(self, path):
+        disk_path = os.path.realpath(path)
+        return os.path.basename(disk_path)
+
+    def _get_sector_size(self, drive):
+        basename = self._resolve_disk_name(drive)
+        block_device = os.path.realpath("/sys/class/block/" + basename)
+        if not block_device.endswith("block/" + basename):
```

Comment:
I think it would be nice if we can open them once and reuse the values like `try/except` in `hddtemp`.

Reply:
memoized it